facebook/src/com/facebook/internal/Validate.java

```diff
@@ -188,16 +188,21 @@ public static void hasFacebookActivity(Context context, boolean shouldThrow) {
         }
     }
 
-    public static void hasContentProvider(Context context) {
+    public static boolean hasContentProvider(Context context) {
```

Comment:
Having this not throw does not match the pattern of the other methods in this file nor a typical validate or assert class. 
Is there a reason why you changed this to return a boolean?